relayer/src/foreign_client.rs

```diff
@@ -398,31 +402,21 @@ impl ForeignClient {
     /// Wrapper for build_update_client_with_trusted.
     pub fn build_update_client(
         &self,
-        target_height: Height,
+        target_height: NonZeroHeight,
     ) -> Result<Vec<Any>, ForeignClientError> {
-        self.build_update_client_with_trusted(target_height, Height::zero())
+        let trusted_height = self.latest_consensus_state_height()?;
+        self.build_update_client_with_trusted(target_height, trusted_height)
     }
 
     /// Returns a vector with a message for updating the client to height `target_height`.
     /// If the client already stores consensus states for this height, returns an empty vector.
     pub fn build_update_client_with_trusted(
         &self,
-        target_height: Height,
-        trusted_height: Height,
+        target_height: NonZeroHeight,
+        trusted_height: NonZeroHeight,
     ) -> Result<Vec<Any>, ForeignClientError> {
-        // Wait for source chain to reach `target_height`
-        while self.src_chain().query_latest_height().map_err(|e| {
```

Comment:
This is probably still necessary, in particular for packet relaying algorithms.

Reply:
I have added a `wait_src_chain_reach_height()` function to wait for the source chain to reach the target height. So far the only places that I know need waiting is in `RelayPath` and the command line.